### PR TITLE
[Fix] PHP Notice Bad Index on jsCallback

### PR DESCRIPTION
**Pull request by ibelar**
Refactor jsCallback::getAjaxec method.
- Returning a View is not dependant of index[0] anymore when in array.

Fix: #1010

### CODE REVIEW

File: src/jsCallback.php [hunk 1]
```diff
@@ -122,11 +122,13 @@ public function set($callback, $args = [])
...
      *
-     * @param array|jsExpressionable $ajaxec  Array of jsExpressionable
-     * @param string                 $msg     General message, typically won't be displayed
-     * @param bool                   $success Was request successful or not
+     * @param array|jsExpressionable $ajaxec Array of jsExpressionable
+     * @param string $msg General message, typically won't be displayed
+     * @param bool $success Was request successful or not
      *
-     * @return [type] [description]
+     * @return void [type] [description]
```
Comment (mvorisek) [inline src/jsCallback.php]: `[type] [description]` is probably unintended.

If nothing is expected to be returned, do not add phpdoc at all and add return type `: void` to the method header directly.

File: src/jsCallback.php [hunk 2]
```diff
@@ -188,4 +175,46 @@ public function getURL($mode = 'callback')
...
+
+        return $action;
+    }
+
+    /**
+     * Render View into modal.
+     *
+     * @param View $response
+     *
+     * @return jsExpression
```
Comment (mvorisek) [inline src/jsCallback.php]: Remove phpdoc and force type in the code directly.

File: src/jsCallback.php [hunk 3]
```diff
@@ -188,4 +175,46 @@ public function getURL($mode = 'callback')
...
+     * @throws \atk4\core\Exception
+     */
+    private function _jsRenderIntoModal($response)
+    {
+        $html = '<div class="ui modal"> <i class="close icon"></i>  <div class="content atk-content"> ';
```
Comment (mvorisek) [inline src/jsCallback.php]: Quite hardcoded, is possible to render this using standard `View` components?

File: src/jsCallback.php (hunk 1 above)
Comment (ibelar) [reply src/jsCallback.php]: yes, will remove it.

File: src/jsCallback.php (hunk 3 above)
Comment (ibelar) [reply src/jsCallback.php]: in fact, why not passing Modal directly. Then developper would have total control of Modal to pass in. I will make some change here.

**Comment by mvorisek** [comment]
@ibelar Please rebase on the latest `develop`.

File: src/jsCallback.php (hunk 3 above)
Comment (mvorisek) [reply src/jsCallback.php]: Very good!

File: src/jsCallback.php (hunk 2 above)
Comment (mvorisek) [reply src/jsCallback.php]: I fixed the merge conflicts and fixed this as well.

**Comment by mvorisek** [comment]
@ibelar Behat tests / TravisCI needs to be updated for commit https://github.com/atk4/ui/pull/1017/commits/ae8c38c4c1d99413fed44cf17ab9c5df7a0f878d

**Comment by ibelar** [comment]
@mvorisek - Selenium/Behat test are fixed.

**Pull request merged**